docs/consensus/consensus_chain.md

```diff
@@ -109,12 +109,10 @@ Justifications contain a set of all PoT checkpoints since the parent block up to
 ## Fast sync
 
 1. Obtain segment headers from DSN: 1-4 from [sync from DSN implementation](#Synchronization)
-2. If the last segment index is greater than 2 - download the last two segments from DSN otherwise exit [Fast sync](#fast-sync) and continue with either [sync from DSN implementation](#Synchronization) or [Substrate Sync](#substrate-sync) .
-3. Reconstruct blocks from the two downloaded segments.
-4. Import the last block from the second last segment into the blockchain DB bypassing the blockchain checks of missing parent block.
-5. Import the state using [Substrate Sync](#substrate-sync)  for the newly imported block.
-6. Import and execute all blocks from the last segment in regular mode.
-7. Pass the control to [sync from DSN implementation](#Synchronization). It will either download the new archived segment if any or pass the control to [Substrate Sync](#substrate-sync).
+2. If the last segment index is greater than 2 - download the last two segments from DSN otherwise exit [Fast sync](#fast-sync) and continue with either [sync from DSN implementation](#Synchronization) or [Substrate Sync](#substrate-sync) . Reconstruct blocks from the two downloaded segments. Note: we need only a single last block from the second last segment and we can optimize the solution if acquire this block from Substrate network instead.
```

Comment:
This is a better version, but I meant slightly differently:
```suggestion
2. Download and reconstruct all blocks from the last segment of archived history
    * Note: In most cases it'll be necessary to download second last segment as well due to the first block being partially included in latest segment
```

The idea is that we should convey what we are trying to achieve. And we want all blocks from the last segment. The fact that segment before that is also needed is a technical implementation detail.